### PR TITLE
Add Payment error dialog and retry payment support

### PR DESCRIPTION
**Pull request by malinajirka**
Parent issue #3726 

This PR adds UI for Failed Payment state and adds support for "retry". A generic "Payment failed please try again" message is shown for all types of errors - the messaging will be updated in future PRs.

To Test - Make sure to test these changes with a site with WCPay in devmode (or live mode, but only hw reader is supported in live mode)
1. Open app settings
2. Tap on "Manage card reader"
3. (optional) Check "use simulated reader"
4. Tap on "Connect card reader"
5. Go back - open a detail of an order in **USD currency**
6. Tap on "Collect payment"
7. Check that the flow works and the payment succeeds
8. Dismiss the dialog - "print/send receipt" buttons are not implemented yet
9. Tap on "Collect payment" again
10. Turn on Airplane mode
11. Notice the payment fails and an error dialog is shown
12. Turn of airplane mode
13. Wait until the device has a connection and tap on "Retry"
14. Notice the payment flow resumes and completes succesfully

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -599,18 +599,20 @@
...
     <string name="card_reader_payment_collect_payment_header" translatable="false">@string/card_reader_collect_payment</string>
     <string name="card_reader_payment_processing_payment_header" translatable="false">@string/card_reader_collect_payment</string>
     <string name="card_reader_payment_capturing_payment_header" translatable="false">@string/card_reader_collect_payment</string>
     <string name="card_reader_payment_completed_payment_header">Payment successful</string>
+    <string name="card_reader_payment_payment_failed_header">Payment failed</string>
 
     <string name="card_reader_payment_collect_payment_state">Reader is ready</string>
     <string name="card_reader_payment_processing_payment_state">Processing payment</string>
     <string name="card_reader_payment_capturing_payment_state">Capturing payment</string>
+    <string name="card_reader_payment_failed_unexpected_error_state">Payment failed please try again</string>
```
Comment (nbradbury) [inline WooCommerce/src/main/res/values/strings.xml]: This seems like a run-on sentence. Perhaps "Payment failed. Please try again." or "Payment failed, please try again".

Comment (malinajirka) [reply WooCommerce/src/main/res/values/strings.xml]: 👍 Fixed in 649a2f2895c7dfbda8d734b3e23c77ad308b83bf. Thanks

**Comment by nbradbury** [comment]
This doesn't need to be considered for this PR, but wanted to mention it's a little jarring to tap retry and the dialog changes size. Maybe we can keep the dialog the same size, with the progress centered?

ezgif-2-f3bf797f6a86

**Comment by malinajirka** [comment]
Thanks for the review @nbradbury !

> This doesn't need to be considered for this PR, but wanted to mention it's a little jarring to tap retry and the dialog changes size. Maybe we can keep the dialog the same size, with the progress centered?

💯 agree, I actually already mentioned it to Garance and we plan to make sure all the dialogs are the same size. I have an item "Polish UI" on the github issues during which I plan to finish these small UI issues.

**Review by nbradbury — APPROVED** [review]
:shipit:

**Pull request merged**